mundane-io/src/main/scala/com/ambiata/mundane/io/LocalDirectory.scala

```diff
@@ -0,0 +1,193 @@
+package com.ambiata.mundane.io
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.path._
+import java.io._
+import java.net.URI
+import scalaz._, Scalaz._, effect._
+
+/**
+ * 'LocalDirectory' represents a directory that exists on a local file system
+ */
+class LocalDirectory private (val path: Path) extends AnyVal {
+  def toLocalPath: LocalPath =
+    LocalPath(path)
+
+  def toFile: File =
+    new File(path.path)
+
+  def exists: RIO[Boolean] = RIO.safe[Boolean] {
+    val file = toFile
+    file.exists && file.isDirectory
+  }
+
+  def onExists[A](success: => RIO[A], missing: => RIO[A]): RIO[A] =
+    exists >>= (e =>
+      if (e) success
+      else   missing
+    )
+
+  def whenExists(thunk: => RIO[Unit]): RIO[Unit] =
+    onExists(thunk, RIO.unit)
+
+  def doesExist[A](error: String, thunk: => RIO[A]): RIO[A] =
+    onExists(thunk, RIO.failIO(error))
+
+  def doesNotExist[A](error: String, thunk: => RIO[A]): RIO[A] =
+    onExists(RIO.failIO(error), thunk)
+
+/**
+  This needs to list recursively and delete from the leaf up.
+  i.e.
+  .
+  └── a
+      ├── b
+      │   ├── bar
+      │   └── c
+      └── foo
+  Delete in this order: (bar -> c -> b -> foo -> a)
+  */
+  def delete: RIO[Unit] = for {
+    f <- RIO.safe(Option(path.toFile.listFiles).cata(_.toList, List()))
+    _ <- f.traverse(p => (toLocalPath | Component.unsafe(p.getName)).delete)
```

Comment:
Very minor quibble but it would be nice to get rid of as any unsafe calls as possible, so can this just be `(toLocalPath /- p.getName)`